data/crac-creation/crac-creator-csa-profiles/src/test/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/CsaProfileCracCreatorTest.java

```diff
@@ -0,0 +1,246 @@
+
+import com.farao_community.farao.data.crac_api.Contingency;
+import com.farao_community.farao.data.crac_api.NetworkElement;
+import com.farao_community.farao.data.crac_creation.creator.api.parameters.CracCreationParameters;
+import com.farao_community.farao.data.crac_creation.creator.csa_profile.CsaProfileCrac;
+import com.farao_community.farao.data.crac_creation.creator.csa_profile.importer.CsaProfileCracImporter;
+import com.google.common.base.Suppliers;
+import com.powsybl.computation.local.LocalComputationManager;
+import com.powsybl.iidm.network.ImportConfig;
+import com.powsybl.iidm.network.Network;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.InputStream;
+import java.nio.file.Paths;
+import java.time.OffsetDateTime;
+import java.util.Iterator;
+import java.util.Properties;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class CsaProfileCracCreatorTest {
+
+    @Test
+    public void testCreateCracTestConfigurationTC1v29Mar2023() {
+        Properties importParams = new Properties();
+        Network network = Network.read(Paths.get(new File(CsaProfileCracCreatorTest.class.getResource("/TestConfiguration_TC1_v29Mar2023.zip").getFile()).toString()), LocalComputationManager.getDefault(), Suppliers.memoize(ImportConfig::load).get(), importParams);
+
+        CsaProfileCracImporter cracImporter = new CsaProfileCracImporter();
+        InputStream inputStream = getClass().getResourceAsStream("/TestConfiguration_TC1_v29Mar2023.zip");
+        CsaProfileCrac nativeCrac = cracImporter.importNativeCrac(inputStream);
+
+        CsaProfileCracCreator cracCreator = new CsaProfileCracCreator();
+        CsaProfileCracCreationContext cracCreationContext = cracCreator.createCrac(nativeCrac, network, OffsetDateTime.parse("2023-03-29T12:00Z"), new CracCreationParameters());
+
+        assertNotNull(cracCreationContext);
+        assertTrue(cracCreationContext.isCreationSuccessful());
+        assertEquals(2, cracCreationContext.getCrac().getContingencies().size());
+
+        Iterator<Contingency> it = cracCreationContext.getCrac().getContingencies().iterator();
+        Contingency c1 = it.next();
+        assertEquals("c0a25fd7-eee0-4191-98a5-71a74469d36e", c1.getId());
+        assertEquals("TENNET_TSO_CO1", c1.getName());
+        assertEquals(1, c1.getNetworkElements().size());
+        Iterator<NetworkElement> it1 = c1.getNetworkElements().iterator();
+        assertEquals("b18cd1aa-7808-49b9-a7cf-605eaf07b006 + e8acf6b6-99cb-45ad-b8dc-16c7866a4ddc", it1.next().getId());
+
+        Contingency c2 = it.next();
+        assertEquals("493480ba-93c3-426e-bee5-347d8dda3749", c2.getId());
+        assertEquals("ELIA_CO1", c2.getName());
+        assertEquals(1, c2.getNetworkElements().size());
+        Iterator<NetworkElement> it2 = c2.getNetworkElements().iterator();
+        assertEquals("17086487-56ba-4979-b8de-064025a6b4da + 8fdc7abd-3746-481a-a65e-3df56acd8b13", it2.next().getId());
+    }
+
```

Comment:
maybe add a private function to make your code smaller : 
```suggestion
private void assertContingencyEquals(Contingency contingency, String id, String name, Set<NetworkElement> networkElements)
```

Reply:
done